soc: x86: add Lakemont SoC

dcpleung opened the pull request:
This adds a very basic SoC configuration for Intel Lakemont SoC.

dcpleung committed:
soc: x86: add Lakemont SoC

This adds a very basic SoC configuration for Intel Lakemont SoC.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung commented:
This also includes https://github.com/zephyrproject-rtos/zephyr/pull/31198. Will rebase once that is merged.

dcpleung committed:
x86: add kconfig CONFIG_X86_PC_COMPATIBLE

This is an hidden option to indicate we are building for
PC-compatible devices (where there are BIOS, ACPI, etc.
which are standard on such devices).

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
x86: add kconfig CONFIG_X86_MEMMAP

This adds a new kconfig to enable the use of memory map.
This map can be populated automatically if
CONFIG_MULTIBOOT_MEMMAP=y or can be manually defined
via x86_memmap[].

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung commented:
#31198 is merged. So this is ready for review.

jenmwms commented on soc/x86/lakemont/linker.ld:
```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright (c) 2011-2014, Wind River Systems, Inc.
+ *
```
missing Copyright (c) 2021 Intel Corporation

dcpleung replied:
This is a straight copy from `soc/x86/atom/linker.ld` without any modifications, hence no new copyright line.

jenmwms replied:
Oh ok, thank you!

dcpleung commented:
CI failure is related to https://github.com/zephyrproject-rtos/zephyr/issues/32344

nashif commented:
can we have a user of this? maybe in qemu?

dcpleung committed:
boards: x86/qemu: enable CPU flags for MMX/SSE

Tells QEMU to enable CPU flags corresponding to MMX/SSE
kconfigs.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
board: x86: add new board qemu_x86_lakemont

This adds a new board qemu_x86_lakemont for testing
the Lakemont SoC configuration.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
tests: fpu_sharing: allow board qemu_x86_lakemont

This adds qemu_x86_lakemont to the platform allow list
for the FPU sharing tests. Since Lakemont supports SSE3
and SSSE3, it is better to test them also.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung commented:
> can we have a user of this? maybe in qemu?

Added a new QEMU board in last push.

nashif commented on boards/x86/qemu_x86/qemu_x86_lakemont.yaml:
```diff
@@ -0,0 +1,11 @@
+  - xtools
+  - llvm
+testing:
+  default: true
```
make this for now just run the kernel tests

```
  only_tags:
     - kernel
```

dcpleung replied:
Added.
